File: leetcode/bitmasking/1009. Complement of Base 10 Integer/1009. Complement of Base 10 Integer.cpp
class Solution {
public:
    int getBitMask(int n) {
        if(!n) return 1;
        int mask = 1;
        while(mask < n) {
            mask = (mask << 1) | 1;
        }
        return mask;
    }
    int bitwiseComplement(int n) {
        return n ^ getBitMask(n);
    }
};

File: leetcode/bitmasking/1009. Complement of Base 10 Integer/1009. Complement of Base 10 Integer.rs
impl Solution {
    pub fn get_bit_mask(n: i32) -> i32 {
        if n == 0 {
            return 1;
        }
        let mut mask = 1;
        while mask < n {
            mask = (mask << 1) | 1;
        }
        return mask;
    }
    pub fn bitwise_complement(n: i32) -> i32 {
        return n ^ Self::get_bit_mask(n);
    }
}